src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -1008,6 +1008,24 @@ final class TrieMap[K, V] private (r: AnyRef, rtupd: AtomicReferenceFieldUpdater
     r.cachedSize(this)
   }
 
+  private[collection] def filterInPlaceImpl(p: (K, V) => Boolean): this.type = {
+    val it = readOnlySnapshot().iterator
```

Comment:
could this use `iterator` (which only calls `readOnlySnapshot()` if `nonReadOnly`)?

Reply:
if we're trying to call `filterInPlace` on a read-only snapshot, something has gone horribly, horribly wrong.

regardless, the implementation needs to be moved up to the `concurrent.Map` trait (see https://github.com/scala/bug/issues/12443#issuecomment-898899755), so it's just going to call `iterator` without `readOnlySnapshot()` after we push that change anyway. we're currently figuring out how not to duplicate the test for the two implementations before pushing the change though